frontend/utils/tabVisibility.ts

```
@@ -0,0 +1,100 @@
+        return false;
+      }
+
+      if (!rule.domains || rule.domains.length === 0) {
+        return rule.defaultVisible ?? true;
+      }
+
+      if (!userDomain) {
+        return rule.defaultVisible ?? false;
+      }
+
+      const normalizedUserDomain = userDomain.toLowerCase().trim();
```

Biome (2.4.10)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)